modules/mod_ginger_base/controllers/controller_rest_resources.erl

```diff
@@ -100,21 +100,30 @@ process_post(Req, State = #state{mode = collection}) ->
     {Body, Req1} = wrq:req_body(Req),
     Data = jsx:decode(Body, [return_maps, {labels, atom}]),
     Props = lists:foldl(fun post_props/2, [], maps:to_list(Data)),
-    {ok, Id} = m_rsc:insert(Props, Context),
-    %% Create edges
-    lists:foreach(
-      fun (Edge) ->
-              {ok, PredicateId} = m_rsc:name_to_id(maps:get(predicate, Edge), Context),
-              {ok, _EdgeId} = m_edge:insert(Id, PredicateId, maps:get(object, Edge), Context)
-
-      end,
-      maps:get(edges, Data, [])
-     ),
-    %% Set "Location" header
-    Location = "/data/resources/" ++ erlang:integer_to_list(Id),
-    Req2 = wrq:set_resp_headers([{"Location", Location}], Req1),
-    %% Done
-    {{halt, 201}, Req2, State}.
+    try m_rsc:insert(Props, Context) of
+        {ok, Id} ->
+            %% Create edges
+            lists:foreach(
+              fun (Edge) ->
+                      {ok, PredicateId} = m_rsc:name_to_id(maps:get(predicate, Edge), Context),
+                      {ok, _EdgeId} = m_edge:insert(Id, PredicateId, maps:get(object, Edge), Context)
+              end,
+              maps:get(edges, Data, [])
+             ),
+            %% Set "Location" header
+            Location = "/data/resources/" ++ erlang:integer_to_list(Id),
+            Req2 = wrq:set_resp_headers([{"Location", Location}], Req1),
+            %% Done
+            {{halt, 201}, Req2, State}
+    catch
+        throw:{{error, nocategory}, _} ->
+            lager:error("Category is missing from the request body", []),
+            {{halt, 400}, Req1, State};
+        throw:{{error, eacces}, _} ->
+            {{halt, 403}, Req1, State};
+        throw:{_} ->
+            {{halt, 500}, Req1, State}
```

Comment:
If you leave out this case the server crashes and so returns a 500 anyway. So just remove it. 😉